### PR TITLE
Fixed #26555 -- Gave deconstructible objects a higher priority during serialization

### PR DESCRIPTION
**Pull request by MarkusH**
Ticket: https://code.djangoproject.com/ticket/26555

### CODE REVIEW

File: django/db/migrations/serializer.py [hunk 1]
```diff
@@ -329,6 +329,18 @@ def serializer_factory(value):
         # tuple.
         value = value.__reduce__()[1][0]
 
+    if isinstance(value, models.Field):
+        return ModelFieldSerializer(value)
+    if isinstance(value, models.manager.BaseManager):
+        return ModelManagerSerializer(value)
+    # Anything that knows how to deconstruct itself.
```
Comment (charettes) [inline django/db/migrations/serializer.py]: Shouldn't this comment be moved before the `hasattr(value, 'deconstruct')`.

Comment (MarkusH) [inline django/db/migrations/serializer.py]: Indeed!

**Comment by charettes** [comment]
Big +1 from me, this will make deconstructible objects way more useful. See this [SO](https://stackoverflow.com/questions/35632376/django-migrations-and-deconstructible-string/35634504#35634504) for a good example of what this will solve.

**Comment by mjtamlyn** [comment]
+1

**Pull request merged**